app/Http/Controllers/Admin/CategoryController.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace App\Http\Controllers\Admin;
+
+use Illuminate\Http\Request;
+use App\Http\Controllers\Controller;
+use App\Models\Category;
+use Session;
+use App\Http\Requests\CategoryRequest;
+
+class CategoryController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $data = Category::orderBy('id', 'DESC')->paginate();
+        return view('admin.categories.index', compact('data', $data));
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        return view('admin.categories.create');
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param \Illuminate\Http\Request $request of category
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function store(CategoryRequest $request)
+    {
+        $category = new Category($request->all());
+        $category ->save();
+        Session::flash(trans('messages.success'), trans('messages.category_create_success'));
```

Comment:
the key of flash session, no need to use `trans`. Because it's fixed, not changed by language